src/Infrastructure/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -32,6 +33,7 @@ public static IApplicationBuilder UseInfrastructure(this IApplicationBuilder app
             app.UseRouting();
             app.UseCors("CorsPolicy");
             app.UseAuthentication();
+            app.UseMiddleware<TenantMiddleware>();
```

Comment:
Not sure if the TenantMiddleware should maybe go to the the Bootstrapper project to be consistent with the other middlewares? I think it should from a "clean architecture" perspective...

Reply:
There is a separate AddMiddlewares method in the Infra Extension. This can go there.

Reply:
Oh ok I didn't realize the middlewares are already in the infra project... yeah better put it all together there then.

Reply:
I would be more interested in understanding what happens if this middleware is commented out. Will the entire flow break, or is there a fallback? @frankyjquintero

Reply:
ok, so we get this - "exception": "No service for type 'DN.WebApi.Infrastructure.Middlewares.TenantMiddleware' has been registered.",
if the middleware is not there in the pipeline. It should probably be possible to leverage this approach and make this an optional multitenant application maybe down the road.